From 30a192e4409fcf69b4104b6390a6de1550a5d0dc Mon Sep 17 00:00:00 2001
From: Eduardo Habkost <ehabkost@redhat.com>
Date: Tue, 20 Aug 2013 15:09:52 +0200
Subject: [PATCH 073/212] pc: Haswell doesn't have rdtscp on rhel6.x

RH-Author: Eduardo Habkost <ehabkost@redhat.com>
Message-id: <1377011392-9336-8-git-send-email-ehabkost@redhat.com>
Patchwork-id: 53615
O-Subject: [RHEL7 PATCH 7/7] pc: Haswell doesn't have rdtscp on rhel6.x
Bugzilla: 918907
RH-Acked-by: Markus Armbruster <armbru@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>

Bugzilla: 918907
Upstream status: not applicable

This adds compat code to disable rdtscp on the rhel6.x machine-types, to
match the features found on RHEL-6.

Signed-off-by: Eduardo Habkost <ehabkost@raisama.net>
---
 hw/i386/pc_piix.c | 2 ++
 1 file changed, 2 insertions(+)

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 hw/i386/pc_piix.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/hw/i386/pc_piix.c b/hw/i386/pc_piix.c
index 6a1cc45..1d06c76 100644
--- a/hw/i386/pc_piix.c
+++ b/hw/i386/pc_piix.c
@@ -1040,6 +1040,8 @@ static void pc_init_rhel650(MachineState *machine)
              0);
     x86_cpu_compat_set_features("SandyBridge", FEAT_8000_0001_EDX,
                                 0, CPUID_EXT2_RDTSCP);
+    x86_cpu_compat_set_features("Haswell", FEAT_8000_0001_EDX,
+                                0, CPUID_EXT2_RDTSCP);
     pc_init_rhel700(machine);
 }
 
-- 
1.7.1

